src/status_im2/common/scan_qr_code/view.cljs

```diff
@@ -0,0 +1,239 @@
+       (i18n/label :t/ensure-qr-code-is-in-focus-to-scan)]]]))
+(defn render-camera
+  [show-camera? qr-view-finder camera-ref on-read-code show-holes?]
+  (when (and show-camera? (:x qr-view-finder))
+    [:<>
+     [rn/view {:style style/camera-container}
+      [camera-kit/camera
+       {:ref            #(reset! camera-ref %)
+        :style          style/camera-style
+        :camera-options {:zoomMode :off}
+        :scan-barcode   true
+        :on-read-code   on-read-code}]
+     ]
+     [hole-view/hole-view
+      {:style style/hole
+       :holes (if show-holes?
+                [(merge qr-view-finder
+                        {:borderRadius 16})]
+                [])}
+      [blur/view
+       {:style            style/absolute-fill
+        :blur-amount      10
+        :blur-type        :transparent
+        :overlay-color    colors/neutral-80-opa-80
+        :background-color colors/neutral-80-opa-80}]]]))
+
+(defn- check-qr-code-data
+  [event]
+  (let [connection-string        (string/trim (oops/oget event "nativeEvent.codeStringValue"))
+        valid-connection-string? (sync-utils/valid-connection-string? connection-string)]
+    (if valid-connection-string?
+      (rf/dispatch [:syncing/input-connection-string-for-bootstrapping connection-string])
+      (rf/dispatch [:toasts/upsert
+                    {:icon           :i/info
+                     :icon-color     colors/danger-50
+                     :override-theme :light
+                     :text           (i18n/label :t/error-this-is-not-a-sync-qr-code)}]))))
+
+
+(defn- scan-qr-code-view
+  [qr-view-finder]
+  [:<>
+   [rn/view {:style style/scan-qr-code-container}]
+   (when (empty? @qr-view-finder)
+     [qr-scan-hole-area qr-view-finder])
+   (if (and @preflight-check-passed?
+            @camera-permission-granted?
+            (boolean (not-empty @qr-view-finder)))
+     [viewfinder @qr-view-finder]
+     [camera-and-local-network-access-permission-view])])
+
+(defn view
+  [header]
+  (let [insets         (safe-area/get-insets)
+        qr-view-finder (reagent/atom {})]
+    (fn []
+      (let [camera-ref                       (atom nil)
+            read-qr-once?                    (atom false)
+            ;; The below check is to prevent scanning of any QR code
+            ;; when the user is in syncing progress screen
+            user-in-syncing-progress-screen? (= (rf/sub [:view-id]) :syncing-progress)
+            on-read-code                     (fn [data]
+                                               (when (and (not @read-qr-once?)
+                                                          (not user-in-syncing-progress-screen?))
+                                                 (reset! read-qr-once? true)
+                                                 (js/setTimeout (fn []
+                                                                  (reset! read-qr-once? false))
+                                                                3000)
```

Comment:
Probably we need to call `js/clearTimeout` when component unmounts

Reply:
thanks for suggestion, I have added it